src/BuildScriptGenerator/VersionProviderHelper.cs

```diff
@@ -55,5 +40,58 @@ internal static IEnumerable<string> GetVersionsFromDirectory(string versionsDir)
             versions.Sort();
             return versions.Select(v => v.ToString());
         }
+
+        /// <summary>
+        ///     This method is specifically for obtaining Major.Minor version (example: 1.16)
+        ///     specifically for Golang since it does not have use semantic-versioning
+        ///     As a result we use RegEx extract version instead of SemVer.
+        /// </summary>
+        internal static IEnumerable<string> GetMajorMinorVersionsFromDirectory(string versionsDir)
+        {
+            IEnumerable<DirectoryInfo> versionDirectories = GetVersionDirectories(versionsDir);
+            var versions = new List<string>();
+            foreach (var versionDir in versionDirectories)
+            {
+                /* Regex will match:
+                    * major.minor
+                    *      where major & minor contains any number of digits
+                    * Valid examples:
+                    *      1.16, 1.17, 0.1, 123.456
+                    * Invalid examples:
+                    *      1.16.1, 1.2.3, 1.2a
+                */
+                Regex regex = new Regex(@"^[0-9]+\.[0-9]+$");
+                string version = versionDir.Name;
+                Match match = regex.Match(version);
+                if (match.Success)
+                {
+                    versions.Add(version);
+                }
+            }
+
+            return versions;
+        }
+
+        private static IEnumerable<DirectoryInfo> GetVersionDirectories(string versionsDir)
+        {
+            var listOptions = new EnumerationOptions()
+            {
+                RecurseSubdirectories = false,
+                IgnoreInaccessible = false,
+            };
+
+            IEnumerable<DirectoryInfo> versionDirectories;
+            try
+            {
+                versionDirectories = Directory.EnumerateDirectories(versionsDir, "*", listOptions)
+                    .Select(versionDir => new DirectoryInfo(versionDir));
+            }
+            catch (IOException)
+            {
+                return (IEnumerable<DirectoryInfo>)Enumerable.Empty<string>();
```

Comment:
Is the casting necessary here? Any reason we can't just do the following?

```cs
return Enumerable.Empty<DirectoryInfo>();
```

Reply:
Update has been made.